DependencyInjection/DoctrineMongoDBExtension.php

```diff
@@ -427,6 +396,47 @@ private function loadMessengerServices(ContainerBuilder $container): void
         $loader->load('messenger.xml');
     }
 
+    /** @param array<string, mixed> $config */
+    private function loadEntityValueResolverServices(ContainerBuilder $container, FileLoader $loader, array $config): void
+    {
+        // available in Symfony 6.2 and higher
+        if (! class_exists(EntityValueResolver::class)) {
+            return;
+        }
+
+        $loader->load('value_resolver.xml');
+
+        if (! class_exists(ExpressionLanguage::class)) {
+            $container->removeDefinition('doctrine_mongodb.odm.entity_value_resolver.expression_language');
+        }
+
+        $controllerResolverDefaults = [];
+
+        if (! $config['controller_resolver']['enabled']) {
+            $controllerResolverDefaults['disabled'] = true;
+        }
+
+        if (! $config['controller_resolver']['auto_mapping']) {
+            $controllerResolverDefaults['mapping'] = [];
+        }
+
+        if ($controllerResolverDefaults === []) {
+            return;
+        }
+
+        $container->getDefinition('doctrine_mongodb.odm.entity_value_resolver')->setArgument(2, (new Definition(MapEntity::class))->setArguments([
+            null,
+            null,
+            null,
+            $controllerResolverDefaults['mapping'] ?? null,
+            null,
+            null,
+            null,
+            null,
+            $controllerResolverDefaults['disabled'] ?? false,
+        ]));
```

Comment:
Does the container support named arguments? That might make this a little easier to read.